src/client/src/orangehrmBuzzPlugin/components/VideoFrame.vue

```diff
@@ -0,0 +1,72 @@
+<!--
+/**
+ * OrangeHRM is a comprehensive Human Resource Management (HRM) System that captures
+ * all the essential functionalities required for any enterprise.
+ * Copyright (C) 2006 OrangeHRM Inc., http://www.orangehrm.com
+ *
+ * OrangeHRM is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+ -->
+
+<template>
+  <div class="orangehrm-buzz-video">
+    <iframe
+      :src="embedURL"
+      frameborder="0"
+      class="orangehrm-buzz-video-frame"
+      title="OrangeHRM buzz video frame"
+      allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
+      allowfullscreen
+    ></iframe>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'VideoFrame',
+  props: {
+    videoSrc: {
+      type: String,
+      required: true,
+    },
+  },
+  computed: {
+    embedURL() {
```

Comment:
shall we move these logic & validation to BE?
:thinking:

Reply:
Sure, but we will possibly need 2 APIs.

> 1. For validation of user input URL
> 2. To extract video ID of the user input URL (we need only the video ID for embed frame and since there are multiple formats of URL we need a regex to match pattern and extract the ID)